transport/http2_client.go

```diff
@@ -353,7 +353,7 @@ func (t *http2Client) newStream(ctx context.Context, callHdr *CallHdr) *Stream {
 
 // NewStream creates a stream and registers it into the transport as "active"
 // streams.
-func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (_ *Stream, err error) {
+func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr, md metadata.MD) (_ *Stream, err error) {
```

Comment:
Could we add the metadata into the `CallHdr` instead to avoiding proliferating new parameters?

Reply:
Good call! I'm on it.